Docker and Docker Compose update for Windows and new documents for Linux (#607)
OctopusDeploy/docs: docs/installation/octopus-in-container/docker-compose-linux.md

---

> title: Deploying Octopus Server Linux Container using Docker Compose

Reviewer: The deeper a doc is nested the more scrunched up the titles get in the TOC sidebar on the site. Does the title still work if we change it to:
```suggestion
title: Octopus Linux Container with Docker Compose
```

description: A fully self-contained SQL Server and Octopus Server
position: 3
---

> If you're looking to run an Octopus Deploy Windows container or a Tentacle Windows Container, please refer to the [Docker Compose Windows](/docs/installation/octopus-in-container/docker-compose-windows.md) documentation.

Reviewer: Slight tweak:
```suggestion
If you want to run an Octopus Deploy Windows container or a Tentacle Windows Container, please refer to the [Docker Compose Windows](/docs/installation/octopus-in-container/docker-compose-windows.md) documentation.
```

For evaluation purposes you may want to run a stand-alone SQL Server instance alongside the Octopus server. For this scenario, you can leverage [Docker Compose](https://docs.docker.com/compose/overview/) to spin up and manage a multi-container Docker application as a single unit.

> The following example is a simple `docker-compose.yml` file combining an SQL Server instance along with a dependent Octopus server:

Reviewer: ```suggestion
The following example is a simple `docker-compose.yml` file combining an SQL Server instance with a dependent Octopus server:
```

ACCEPT_EULA=N

# Use of this Image means you must accept the Octopus Deploy Eula found here: https://octopus.com/company/legal
ACCEPT_OCTOPUS_EULA=N

# Unique Server Node Name - If left empty will default to the machine Name
OCTOPUS_SERVER_NODE_NAME=

# Database Connection String. If using database in sql server container, it is highly recommended to change the password.
DB_CONNECTION_STRING=Server=db,1433;Database=OctopusDeploy;User=sa;Password=N0tS3cr3t!

# Your License key for Octopus Deploy. If left empty, it will try and create a free license key for you
OCTOPUS_SERVER_BASE64_LICENSE=

# Octopus Deploy uses a master key for encryption of your databse. If you're using an external database that's already been setup for Octopus Deploy, you can supply the master key to use it. If left blank, a new master key will be generated with the database creation.
MASTER_KEY=

# The API Key to set for the administrator. If this is set and no password is provided then a service account user will be created. If this is set and a password is also set then a standard user will be created.
ADMIN_API_KEY=
```

> In this case, we are specifying the `sa` password that is used when starting the sql container and for db connectivity from the Octopus Server. We have also provided the Octopus admin credentials and set a host port mapping to port `8080` so we can access the server externally.

Reviewer: ```suggestion
In this case, we are specifying the `sa` password that is used when starting the SQL container and for db connectivity from the Octopus Server. We have also provided the Octopus admin credentials and set a host port mapping to port `8080` so we can access the server externally.
```

> In this case, we are specifying the `sa` password that is used when starting the sql container and is used for db connectivity from the Octopus Server. We have also provided the Octopus admin credentials and set a host port mapping to port `8080` so we can access the server externally.

Reviewer: ```suggestion
In this case, we are specifying the `sa` password that is used when starting the sql container and for db connectivity from the Octopus Server. We have also provided the Octopus admin credentials and set a host port mapping to port `8080` so we can access the server externally.
```

Start both containers by running:

```
docker-compose --project-name Octopus up -d
```

When both containers are healthy, you can browse directly to `http://localhost` from your host machine.

> 

Reviewer: ```suggestion

## Upgrade with Docker Compose

```
